File: dataprovider-retrofit/src/main/java/ch/srg/dataProvider/integrationlayer/utils/UserAgentInterceptor.java
package ch.srg.dataProvider.integrationlayer.utils;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class UserAgentInterceptor implements Interceptor {

    private String userAgent;

    public UserAgentInterceptor(String userAgent) {
        this.userAgent = userAgent;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request original = chain.request();

        Request.Builder builder = original.newBuilder().header("User-Agent", userAgent);

        Request request = builder.build();
        return chain.proceed(request);
    }
}

File: dataprovider-retrofit/src/main/java/ch/srg/dataProvider/integrationlayer/utils/UserAgentInterceptor.kt
package ch.srg.dataProvider.integrationlayer.utils

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * User agent interceptor
 *
 * @property userAgent The user agent to set to OkHttp.
 */
class UserAgentInterceptor(private val userAgent: String) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val original = chain.request()
        val builder = original.newBuilder().header("User-Agent", userAgent)
        val request = builder.build()
        return chain.proceed(request)
    }
}
